### PR TITLE
feat(ip): add intellectual properties api domain

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

Add a new api domain, Intellectual properties
Feature-flagged.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by lodmfjord — APPROVED** [review]
Change to codeowners LGTM.

File: libs/api/domains/intellectual-properties/src/lib/intellectualProperties.service.ts [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    return object
+  }
+
+  async getDesignImages(
+    user: User,
+    designId: string,
+  ): Promise<Array<Image> | null> {
+    const response = await this.ipService.getDesignImages(user, designId)
+
+    //shady stuff
```
Comment (thordurhhh) [inline libs/api/domains/intellectual-properties/src/lib/intellectualProperties.service.ts:305]: what a comment 😂

File: libs/api/domains/rights-portal/src/lib/therapy/therapy.resolver.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import {
 } from '@island.is/nest/feature-flags'
 import { TherapyService } from './therapy.service'
 import { PaginatedTherapyResponse } from './models/therapy.model'
+import { PaginatedResponse } from '@island.is/nest/pagination'
```
Comment (thordurhhh) [inline libs/api/domains/rights-portal/src/lib/therapy/therapy.resolver.ts:19]: Remove?

**Review by thordurhhh — APPROVED** [review]
Niiice 🙌

File: libs/api/domains/intellectual-properties/src/lib/intellectualProperties.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+@FeatureFlag(Features.isIntellectualPropertyModuleEnabled)
+@UseGuards(IdsUserGuard, ScopesGuard, FeatureFlagGuard)
+@Audit({ namespace: '@island.is/api/intellectual-properties' })
+export class IntellectualPropertiesResolver {
+  constructor(
+    private readonly ipService: IntellectualPropertiesService,
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+  ) {}
+
+  @Scopes(ApiScope.internal)
```
Comment (saevarma) [inline libs/api/domains/intellectual-properties/src/lib/intellectualProperties.resolver.ts:40]: ~ FYI you can also just apply the `@Scopes` decorator on the class level.

File: libs/api/domains/intellectual-properties/src/lib/models/person.model.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  telephone?: string
+
+  @Field(() => String, { nullable: true })
+  mobilePhone?: string
+
+  @Field({ nullable: true })
+  isForeign?: boolean
+
+  @Field(() => String, { nullable: true })
+  ssn?: string
```
Comment (saevarma) [inline libs/api/domains/intellectual-properties/src/lib/models/person.model.ts:40]: ~ The monorepo favors `nationalId` over `ssn`.

**Review by saevarma — APPROVED** [review]
Approved with optional comments but removed the `automerge` label so you can review them before this is merged.

File: libs/api/domains/intellectual-properties/src/lib/models/person.model.ts (hunk 4 above)
Comment (thorkellmani) [reply libs/api/domains/intellectual-properties/src/lib/models/person.model.ts:40]: ah oops, will fix

**Pull request merged**